budgetportal/models.py

```diff
@@ -768,8 +769,324 @@ def get_expenditure_over_time(self):
                            cuts, self.get_financial_year().slug)
             return None
 
-    def __str__(self):
-        return '<%s %s>' % (self.__class__.__name__, self.get_url_path())
+    def get_adjusted_budget_summary(self):
+        dataset = self.get_adjusted_estimates_expenditure_dataset()
+        if not dataset:
+            return None
+        openspending_api = dataset.get_openspending_api()
+        if not openspending_api:
+            return None
+
+        result_for_type_and_total = openspending_api.aggregate(
+            cuts=[
+                openspending_api.get_financial_year_ref() + ':' + self.get_financial_year().get_starting_year(),
+            ],
+            drilldowns=[
+                openspending_api.get_adjustment_kind_ref(),
+                openspending_api.get_phase_ref(),
+                openspending_api.get_programme_name_ref(),
+                openspending_api.get_department_name_ref()
+            ],
+            order=[openspending_api.get_adjustment_kind_ref()])
+
+        result_for_type_and_total = openspending_api.filter_dept(result_for_type_and_total,
+                                                                 self.name)
+        cells_for_type_and_total = openspending_api.filter_and_aggregate(
```

Comment:
You're right - I've implemented this to make it easier to change the logic since we are using filter and aggr in combination in many places, but it is not intuitive and not likely to be used again in the future. Thanks.